src/coins.h

```diff
@@ -20,6 +20,81 @@
 #include <boost/foreach.hpp>
 #include <unordered_map>
 
+/**
+ * A UTXO entry.
+ *
+ * Serialized format:
+ * - VARINT(coinbase + height * 2)
+ * - the non-spent CTxOut (via CTxOutCompressor)
+ */
+class CCoin
+{
+public:
+    //! whether transaction is a coinbase
+    bool fCoinBase;
+
+    //! unspent transaction outputs; spent outputs are .IsNull()
+    CTxOut out;
+
+    //! at which height this transaction was included in the active block chain
+    uint32_t nHeight;
+
+    //! construct a CCoin from a CTransaction, at a given height
+    CCoin(CTxOut&& outIn, int nHeightIn, bool fCoinBaseIn) : fCoinBase(fCoinBaseIn), out(std::move(outIn)), nHeight(nHeightIn) {}
+    CCoin(const CTxOut& outIn, int nHeightIn, bool fCoinBaseIn) : fCoinBase(fCoinBaseIn), out(outIn), nHeight(nHeightIn) {}
+
+    void Clear() {
```

Comment:
Not going to rename things all over the place now. Maybe I can add a commit at the end that fixes it up later.